packages/editor/src/lib/primitives/intersect.ts

```diff
@@ -243,26 +243,32 @@ export function intersectPolygonPolygon(
 	polygonA: VecLike[],
 	polygonB: VecLike[]
 ): VecLike[] | null {
-	// Create an empty polygon as P
-	const result: VecLike[] = []
+	// Create an empty polygon as result
+	const result: Map<string, VecLike> = new Map()
```

Comment:
Had to add deduping to intersect since polygons that have same points might add these points twice.